x/lockup/keeper/synthetic_lock_test.go

```diff
@@ -0,0 +1,311 @@
+	suite.Require().Equal(moduleLockedCoins.String(), "10stake")
+	accountUnlockableCoins = suite.app.LockupKeeper.GetAccountUnlockableCoins(suite.ctx, addr1)
+	suite.Require().Equal(accountUnlockableCoins.String(), "")
+	accountUnlockingCoins = suite.app.LockupKeeper.GetAccountUnlockingCoins(suite.ctx, addr1)
+	suite.Require().Equal(accountUnlockingCoins.String(), "")
+	accountLockedCoins = suite.app.LockupKeeper.GetAccountLockedCoins(suite.ctx, addr1)
+	suite.Require().Equal(accountLockedCoins.String(), "10stake")
+	locks = suite.app.LockupKeeper.GetAccountLockedPastTime(suite.ctx, addr1, suite.ctx.BlockTime())
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetAccountLockedPastTimeNotUnlockingOnly(suite.ctx, addr1, suite.ctx.BlockTime())
+	suite.Require().Len(locks, 1)
+	locks = suite.app.LockupKeeper.GetAccountUnlockedBeforeTime(suite.ctx, addr1, suite.ctx.BlockTime())
+	suite.Require().Len(locks, 0)
+
+	err = suite.app.LockupKeeper.DeleteSyntheticLockup(suite.ctx, 1, "stakedtovalidator1")
+	suite.Require().NoError(err)
+
+	synthLock, err = suite.app.LockupKeeper.GetSyntheticLockup(suite.ctx, 1, "stakedtovalidator1")
+	suite.Require().Error(err)
+	suite.Require().Nil(synthLock)
+
+	// check accumulation store is correctly updated for synthetic lock
+	accum = suite.app.LockupKeeper.GetPeriodLocksAccumulation(suite.ctx, types.QueryCondition{
+		LockQueryType: types.ByDuration,
+		Denom:         "stakestakedtovalidator1",
+		Duration:      time.Second,
+	})
+	suite.Require().Equal(accum.String(), "0")
+}
+
+func (suite *KeeperTestSuite) TestSyntheticLockupDeleteAllSyntheticLocksByLockup() {
+	suite.SetupTest()
+
+	// lock coins
+	addr1 := sdk.AccAddress([]byte("addr1---------------"))
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	suite.LockTokens(addr1, coins, time.Second)
+
+	// check locks
+	locks, err := suite.app.LockupKeeper.GetPeriodLocks(suite.ctx)
+	suite.Require().NoError(err)
+	suite.Require().Len(locks, 1)
+	suite.Require().Equal(locks[0].Coins, coins)
+
+	err = suite.app.LockupKeeper.CreateSyntheticLockup(suite.ctx, 1, "stakedtovalidator1", false)
+	suite.Require().NoError(err)
+
+	synthLock, err := suite.app.LockupKeeper.GetSyntheticLockup(suite.ctx, 1, "stakedtovalidator1")
+	suite.Require().NoError(err)
+	suite.Require().Equal(*synthLock, types.SyntheticLock{
+		LockId:  1,
+		Suffix:  "stakedtovalidator1",
+		EndTime: time.Time{},
+	})
+
+	err = suite.app.LockupKeeper.DeleteAllSyntheticLocksByLockup(suite.ctx, 1)
+	suite.Require().NoError(err)
+
+	synthLock, err = suite.app.LockupKeeper.GetSyntheticLockup(suite.ctx, 1, "stakedtovalidator1")
+	suite.Require().Error(err)
+	suite.Require().Nil(synthLock)
+}
+
+func (suite *KeeperTestSuite) TestSyntheticLockupDeleteAllMaturedSyntheticLocks() {
+	suite.SetupTest()
+
+	// lock coins
+	addr1 := sdk.AccAddress([]byte("addr1---------------"))
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+	suite.LockTokens(addr1, coins, time.Second)
```

Comment:
Lets refactor this / use general logic. (I feel like there should already be some somewhere in the modules test code?)

Reply:
What you mean by general logic here? LockTokens function is a utility function created to reduce the steps of creating a new lockup along with token mint in case token is not available on the account.